### DISCUSSION TITLE
We should have a local velocity on the RigidBody and CharacterBody.

### DISCUSSION THREAD

**Discussion by anameeguy** [godotengine/godot-proposals#9839] [Engine Core]
### Describe the project you are working on

The beginnings of a game where you control a hover bike type thing. (Very new)

### Describe the problem or limitation you are having in your project

The thing is that I want to dampen the up-down movement of the bike.

### Describe the feature / enhancement and how it helps to overcome the problem or limitation

If local velocity were a thing than I could just dampen the local y velocity.

### Describe how your proposal will work, with code, pseudo-code, mock-ups, and/or diagrams

`body.local_velocity`

### If this enhancement will not be used often, can it be worked around with a few lines of script?

Probably.

### Is there a reason why this should be core and not an add-on in the asset library?

I can't really see it handled elegantly as an add-on or library.

**Comment by Calinou** [comment]
Moving to discussion, as there is no concrete technical implementation provided.
